### ISSUE TITLE
Schémas d'URL

### ISSUE THREAD

**Issue by Spone** [telabotanica/naturadapt#3]
Cette issue pour discuter des schémas d'URL.

Voici une première ébauche pour amorcer la discussion :

# Groupes

Chemin | Verbe    | Action
---------- | ---------- | ----------
`/groups` | `GET` | Liste des groupes
`/groups/new` | `GET` | Formulaire de création d'un groupe
`/groups` | `POST` | Création d'un groupe
`/groups/:slug` | `GET` | Consultation d'un groupe
`/groups/:slug/edit` | `GET` | Formulaire de modification d'un groupe
`/groups/:slug` | `PATCH` / `PUT` | Mise à jour d'un groupe
`/groups/:slug` | `DESTROY` | Suppression d'un groupe

# Annuaire

Chemin | Verbe    | Action
---------- | ---------- | ----------
`/members/:id` | `GET` | Consultation du profil d'un membre
`/group/:slug/members` | `GET` | Liste des membres du groupe
`/group/:slug/members/new` | `GET` | Formulaire d'invitation de membres au groupe
`/group/:slug/members` | `POST` | Ajout des membres
`/members/:slug` | `PATCH` / `PUT` | Mise à jour d'un membre (par un admin)

# Actualités

Chemin | Verbe    | Action
---------- | ---------- | ----------
`/groups/:slug/stories` | `GET` | Liste des actualités du groupe
`/groups/:slug/stories/new` | `GET` | Formulaire de création d'une actu
`/groups/:slug/stories` | `POST` | Création d'une actualité
`/groups/:slug/stories/:slug` | `GET` | Consultation d'une actu
`/groups/:slug/stories/:id/edit` | `GET` | Formulaire de modification d'une actu
`/groups/:slug/stories/:id` ou `/stories/:id` ? | `PATCH` / `PUT` | Mise à jour d'une actu
`/groups/:slug/stories/:id` ou `/stories/:id` ? | `DESTROY` | Suppression d'une actu

(à compléter)

**Comment by zemax** [comment]
Ca te va aussi si l'affichage et l'action d'un formulaire sont sur la même URL, ex :
Chemin | Verbe    | Action
---------- | ---------- | ----------
`/groups/new` | `GET` | Formulaire de création d'un groupe
`/groups/new` | `POST` | Création d'un groupe

Ca permet de traiter les 2 dans la même méthode du controller, afficher les erreurs, etc...

**Comment by Spone** [comment]
C'est pas REST ;) mais ça me va si ça simplifie le code

**Comment by zemax** [comment]
C'est pas une obligation mais c'est l'approche classique en Symfony :

> By default, the form will submit a POST request back to the same controller that renders it.
(https://symfony.com/doc/current/forms.html#handling-form-submissions)

**Comment by Spone** [comment]
Je suis d'accord avec toi, il faut s'adapter à l'approche standard du framework.
